### DISCUSSION TITLE
[Feature] Image Details: Source Identification

### DISCUSSION THREAD

**Discussion by jtamm-red** [immich-app/immich#6906] [Feature Request]
### The feature

I would like to know the source of the uploaded image as I use different mobile devices. The confusion arises when trying to identify the origin. I am interested in finding out the sub-folder, such as Camera, Whatsapp Images, etc.

![image](https://github.com/immich-app/immich/assets/53046846/d5e27199-1bc7-4b18-8728-38871f81711c)

Image details should include information on where it was uploaded from, for example, "Uploaded from Android device (**Xiaomi MI 8**)." Additionally, a tooltip should display the subfolder "**Camera**" information.

For instance, when an image is uploaded from a computer via the web, the information content should state "Uploaded from **web**." Similarly, if uploaded via the command line interface (CLI), the content should indicate "Uploaded from **CLI**."

If the image is uploaded from an iPhone 12, the information should read "Uploaded from **Apple device (iPhone 12)**" with a tooltip displaying **"Photo Library."**

In the case of an image uploaded from a computer using a Samsung S20, the information should state "Uploaded from **Android device (Samsung S20)**" and the tooltip should show **"WhatsApp Videos."**

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by i-code-alot** [comment]
I would love to see this. To expand on this more, I have multiple same devices that I use at once, and would love to be able to configure the Immich app to add a custom tag to photos uploaded from each device ex “iPhone 14 Pro Batman” and “iPhone 14 Pro Master Wayne” (two different iPhone 14 Pros I use for the same account)

**Comment by Sofahamster** [comment]
I would also like to see configurable tags or names for individual Immich app installations.

Ideally, the originating device or upload source should also be available as a workflow condition. That would allow workflows to automatically tag assets, add them to albums, or otherwise process them depending on whether they came from a specific phone, the web interface, or the CLI.

A configurable installation name would be more useful than relying on the device model alone.

This could also help with troubleshooting. I recently had duplicate photos from the Android app and an Android browser, with the browser-uploaded copies missing GPS metadata. Being able to identify the upload source would have made the cleanup so much easier.

**Comment by scr4bble** [comment]
One thing worth considering when implementing this: there are potentially two different pieces of information here:

* **Where the asset was originally uploaded/imported from**
* **Which device backup set the asset belongs to**

The second one is more useful for my use case. When migrating to Immich, I first imported my existing libraries from services like Google Photos and only afterwards enabled mobile backup. As a result, many assets already existed in Immich when the phone encountered them, so the phone did not actually upload them.

I would still like those matching assets to get something like a per-device tag, e.g. `device:galaxy-s25`, because that tells me that the photo **exists/existed on that device and was part of its backup set**, regardless of how the server copy originally got into Immich.

If the backup set later changes or the local file is deleted, I think it could also make sense for that tag to remain as historical information.

Just something to consider alongside tracking the actual upload source.

**Reply by scr4bble** [reply]
The key distinction is that an asset doesn't necessarily need to be **uploaded by the device** to belong to that device's backup set. If backup finds an identical asset already on the server, it could still apply the device tag instead of simply skipping the upload.

Because an asset can have multiple tags, this would nicely cover cases where multiple devices would be able to tag it as "device:X" and you would then see you have/had that image in both of them.
